ts/screens/wallet/AddCardScreen.tsx

```diff
@@ -95,301 +98,201 @@ const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
   body: "wallet.saveCard.contextualHelpContent"
 };
 
-const EMPTY_CARD_HOLDER = "";
-const EMPTY_CARD_PAN = "";
-const EMPTY_CARD_EXPIRATION_DATE = "";
-const EMPTY_CARD_SECURITY_CODE = "";
-
-const INITIAL_STATE: State = {
-  pan: none,
-  expirationDate: none,
-  securityCode: none,
-  holder: none
-};
-
-function getCardFromState(state: State): Option<CreditCard> {
-  const { pan, expirationDate, securityCode, holder } = state;
-  if (
-    pan.isNone() ||
-    expirationDate.isNone() ||
-    securityCode.isNone() ||
-    holder.isNone()
-  ) {
-    return none;
-  }
-
-  const [expirationMonth, expirationYear] = expirationDate.value.split("/");
-
-  if (!CreditCardPan.is(pan.value)) {
-    // invalid pan
-    return none;
-  }
-
-  if (
-    !CreditCardExpirationMonth.is(expirationMonth) ||
-    !CreditCardExpirationYear.is(expirationYear)
-  ) {
-    // invalid date
-    return none;
-  }
-
-  if (!CreditCardCVC.is(securityCode.value)) {
-    // invalid cvc
-    return none;
-  }
+const acceptedCardsPageURL: string = "https://io.italia.it/metodi-pagamento";
 
-  const card: CreditCard = {
-    pan: pan.value,
-    holder: holder.value,
-    expireMonth: expirationMonth,
-    expireYear: expirationYear,
-    securityCode: securityCode.value
+const primaryButtonPropsFromState = (
+  state: CreditCardState,
+  onNavigate: (card: CreditCard) => NavigationNavigateAction
+): ComponentProps<typeof FooterWithButtons>["leftButton"] => {
+  const baseButtonProps = {
+    block: true,
+    primary: true,
+    title: I18n.t("global.buttons.continue")
   };
 
-  return some(card);
-}
+  const card = mergeCardValues(state);
 
-const acceptedCardsPageURL: string = "https://io.italia.it/metodi-pagamento";
-
-class AddCardScreen extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      ...INITIAL_STATE,
-      holder: fromNullable(props.profileNameSurname)
+  if (card.isSome()) {
+    return {
+      ...baseButtonProps,
+      disabled: false,
+      onPress: () => onNavigate(card.value)
+    };
+  } else {
+    return {
+      ...baseButtonProps,
+      disabled: true,
+      onPress: () => undefined
     };
   }
+};
 
-  private openSupportedCardsPage = (): void => {
+const AddCardScreen: React.FC<Props> = props => {
+  const [state, setState] = useState<CreditCardState>({
+    ...INITIAL_CARD_FORM_STATE,
+    holder: fromNullable(props.profileNameSurname)
+  });
+
+  const { present } = useIOBottomSheet(
+    <Body>{I18n.t("wallet.missingDataText")}</Body>,
+    I18n.t("wallet.missingDataCTA"),
+    260
+  );
+
+  const openSupportedCardsPage = (): void => {
```

Comment:
Since it is not related to any property, could it be placed outside the component?
In this way we avoid to create this function on each render